eng/Version.Details.xml

```diff
@@ -282,43 +282,43 @@
       <Uri>https://github.com/dotnet/llvm-project</Uri>
       <Sha>26f8c30340764cfa7fa9090dc01a36c222bf09c1</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NETCore.App.Runtime.win-x64" Version="9.0.0-preview.5.24262.2">
+    <Dependency Name="Microsoft.NETCore.App.Runtime.win-x64" Version="9.0.0-preview.5.24273.1">
       <Uri>https://github.com/dotnet/runtime</Uri>
-      <Sha>073e35e7e32624e0f841d94364d2f28c73662ff7</Sha>
+      <Sha>aabefea9ec7c5717177454cd570e48cc55a535bf</Sha>
```

Comment:
@ViktorHofer, maybe it would be nice to display 'Product Version' (which contains the commit hash) in the readme of packages going in dotnet feeds? e.g. https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet9/NuGet/System.Reflection.Metadata/overview/9.0.0-preview.5.24273.1

Current approach is to download the package, unzip and use some tool like exiftool to get the ProductVersion to see the commit hash:

```sh
$ exiftool lib/net9.0/System.Reflection.Metadata.dll -productversion
Product Version                 : 9.0.0-preview.5.24273.1+aabefea9ec7c5717177454cd570e48cc55a535bf
```

Reply:
the dependencies in this file are updated automatically so you shouldn't need to bump them for the preview4 update

Reply:
The penultimate commit was giving me these errors:

> error CS0246: The type or namespace name 'AssemblyNameInfo' could not be found (are you missing a using directive or an assembly reference?

these version bumps helped (no idea how/why.. 😁).

Reply:
That seems to be related to https://github.com/dotnet/runtime/pull/102489#issuecomment-2125777492.
I suggest we wait until that PR lands.

Reply:
Yep - what happened was Preview4 was rebuilt with newer file versions than Preview5 packages already referenced by the product.  So this part of the repo that builds against .NET 9 without referencing live bits started to get the references from the P4 targeting pack with less API than the P5 packages.  The fix is to update to newer P5 packages.

This is a pretty rare circumstance that arose due to rebuild of P4 - normally we'll be updating the next preview's packages and be past the previous preview's versions by the time we make it public.  Better fixes here would be 1) make all parts of the product build against live bits 2) encode preview in fileversion.  Not sure this will happen often enough to go and do those things.

Reply:
> 1) make all parts of the product build against live bits

This is probably the best option. Sometimes it’s a bit tricky to pull it off due to the subsets inter-dependencies. e.g. ilcompiler from clr subset to depend on libs subset for System.Reflection.Metadata in this case. Not impossible to resolve but it’s not particularly straightforward (considering SRM’s own dependencies on live build of ImmutableCollection, Linq etc.).

Reply:
> @ViktorHofer, maybe it would be nice to display 'Product Version' (which contains the commit hash) in the readme of packages going in dotnet feeds? e.g.

That information is already embedded in the nuspec, isn't it? Some package managers display the SHA. You can always download the nupkg and open it with NuGet Package Explorer or unzip and look at the nuspec manually.

Reply:
> That information is already embedded in the nuspec, isn't it?

Yup nuspec has  `<repository type="git" url="https://github.com/dotnet/runtime" commit="aabefea9ec7c5717177454cd570e48cc55a535bf" />`. The readme file can have the version, I remember seeing them in some readmes (as a last line) in the past, can't find an example. That gives a quick, a-the-glance, reference.

Reply:
This would help when looking at packages on nuget.org, AzDO packaging or via VS package manager? Sounds more like a potential UI improvement for of those components. I don't think the SHA should go there and I would rather prefer to keep the package readme files static.

Correct, some packages have the SHA in the package description. That was done as the nuspec didn't contain the information back then. That should eventually go away.

Reply:
Only AzDo; nuget.org's package explorer parses the package number out of nuspec:

<img width="842" alt="image" src="https://github.com/dotnet/runtime/assets/3840695/8b34ae3b-1b61-4245-9c4b-b592e29c8e18">


Yeah, I just thought an informative line won't hurt if it's doable, but if the intent is to keep the readmes static, understood. 👍

Reply:
Reverted this file and merged main. Thanks for the infos everyone!